ERCs/eip-5732/contracts/CommitToMintImpl.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Author: Zainan Victor Zhou <zzn-ercref@zzn.im>
+// Visit our open source repo: http://zzn.li/ercref
+
+pragma solidity ^0.8.9;
+
+import "hardhat/console.sol";
+import "./IERC5732.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+
+/**
+ * @dev Implementation of the {IERC_COMMIT} interface for the Mint use case.
+ * Assuming "TokenID" represents something intersting that people want to
+ * run to mint for. This reference implementation breakdown the minting
+ * process into two steps:
+ *  Step1. One user calls the `commit` inorder to commit to a minting request
+ *        but the actual `tokenId` is not yet revealed to general public.
+ *  Step2. After sometime, that same user calls the "mint" with the actual
+ *        `tokenId` to mint the token, which reveals the token.
+ *        The mint request also contains the a `secret_sault` in its ExtraData.
+ */
+contract CommitToMintImpl is ERC721, IERC_COMMIT  {
+    mapping(address => bytes32) public commitToMint;
+    event ErcRefImplDeploy(uint256 version, string name, string url);
+    constructor(uint256 _version) ERC721("CommitToMintImpl", "CTMI") {
+        emit ErcRefImplDeploy(_version, "CommitToMintImpl", "http://zzn.li/ercref");
```

Comment:
IPFS content URI is always preferable for anything on-chain, for obvious reasons :)

Reply:
It's a re-direct link placeholder actually. We have not chosen a project domain name. once we chose a project domain name we might want to change to it.